openquakeplatform/openquakeplatform/static/js/hrde_hazard_chart_single.js

```diff
@@ -0,0 +1,200 @@
+            .scale(x)
+            .orient('bottom')
+            .ticks(5);
+    }
+
+    function make_y_axis() {
+        return d3.svg.axis()
+            .scale(y)
+            .orient('left')
+            .ticks(5);
+    }
+
+    // grid lines
+    svg.append('g')
+        .attr('class', 'grid')
+        .attr('transform', 'translate(0,' + height + ')')
+        .call(make_x_axis()
+            .tickSize(-height, 0, 0)
+            .tickFormat('')
+        );
+    svg.append('g')
+        .attr('class', 'grid')
+        .call(make_y_axis()
+            .tickSize(-width, 0, 0)
+            .tickFormat('')
+        );
+    svg.append('path')
+        .data([data])
+        .attr('class', 'line')
+        .attr('d', line);
+    svg.append('g')
+        .attr('class', 'x axis')
+        .attr('transform', 'translate(0,' + height + ')')
+        .call(xAxis)
+        .selectAll("text")
+        .style("text-anchor", "end")
+        .attr("dx", "-.8em")
+        .attr("dy", ".15em")
+        .attr("transform", function(d) {
+            return "rotate(-35)";
+            });
+    svg.append('g')
+        .attr('class', 'x axis')
+        .append('text')
+        .attr('x', width / 2)
+        .attr('y',  (height + margin.bottom)- 17)
+        .attr('text-anchor', 'middle')
+        .style('font-size','14px')
+        .text(AppVars.layerImt);
+    svg.append('g')
+        .attr('class', 'y axis')
+        .call(yAxis)
+        .append('text')
+        .attr('transform', 'rotate(-90)')
+        .attr('y', -70)
+        .attr('x', -20)
+        .attr('dy', '.71em')
+        .style('font-size','12px')
+        .style('text-anchor', 'end')
+        .text(yAxisLable);
+    var legend = d3.select('#chartDialog').append('svg')
+        .attr('height', 25);
+    textTopTitle = svg.append("text")
+        .attr("x", 0)
+        .attr("y", -45)
+        .attr("dy", ".35em")
+        .style("font-weight", "bold")
+        .attr("font-size","14px")
+        .text(AppVars.selectedHazardLayerName);
+    textTopTitle2 = svg.append("text")
+        .attr("x", 0)
+        .attr("y", -30)
+        .attr("dy", ".35em")
+        .style("font-weight", "bold")
+        .attr("font-size","14px")
+        .text(AppVars.selectedLayerValue);
+    textTopLable = svg.append("text")
+        .attr("x", 0)
+        .attr("y", -15)
+        .attr("dy", ".35em")
+        .attr("font-size","12px")
+        .text('Point coordinates (lon, lat [in DD]): '+lng+', '+lat);
+    textBottom = svg.append('text')
+        .attr('x', 0)
+        .attr('y', 335)
+        .attr('dy', '.35em')
+        .text('');
+
+    $('#chartDialog').append('<div id="downloadCurve"><font color="blue">Download Curve</font></div>');
+    $('#downloadCurve').on('hover', function(){
+        $(this).css('cursor', 'pointer');
+    });
+
+    var h = $('#chartDialog').height();
+    h = h + 20;
+
+    $('#chartDialog').css({'height': h+'px'});
+
+    var prob, iml;
+
+    if (curveType === 'spectrum') {
```

Comment:
please see: http://stackoverflow.com/questions/359494/does-it-matter-which-equals-operator-vs-i-use-in-javascript-comparisons